solver/src/http_solver.rs

```diff
@@ -31,6 +30,13 @@ pub struct SolverConfig {
 }
 
 impl SolverConfig {
+    pub fn new(max_nr_exec_orders: u32, time_limit: u32) -> Self {
+        SolverConfig {
+            max_nr_exec_orders,
+            time_limit,
+        }
+    }
```

Comment:
Nit: I'd keep the struct without constructor because it is just plain old data. No need to go through a function to create it.

Reply:
This was just to avoid making the struct fields public (would be required since it is being created from a different place). Would you still prefer the no-ctor version?

Reply:
My rule of thumb is that if it is plain old data with no further invariants then no constructor and public fields instead.